src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -1848,67 +1863,63 @@ public void mySignal(String value) {
 
   @Test
   public void testSignal() {
```

Comment:
Might be worth splitting this test? I feel it's doing too much it's no longer clear what it's validating.
Maybe a test for: queries don't cause thread leaks (for both sticky and non sticky)
Another test for testing signals? Is the intention here to test signaling or test worker querying? This seems to be doing both and might be worth splitting that too.

Reply:
Fixed